src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -442,19 +441,125 @@ private BoundExpression VisitObjectInitializerMember(BoundObjectInitializerMembe
                         if (!isRhsNestedInitializer)
                         {
                             // Rewrite as simple assignment.
-                            var rewrittenRight = VisitExpression(assignment.Right);
+                            var rewrittenRight = VisitExpression(right);
                             result.Add(MakeStaticAssignmentOperator(assignment.Syntax, rewrittenAccess, rewrittenRight, false, assignment.Type, used: false));
                             return;
                         }
 
                         break;
                     }
 
+                case BoundKind.ImplicitIndexerAccess:
+                    if (temps == null)
+                    {
+                        temps = ArrayBuilder<LocalSymbol>.GetInstance();
+                    }
+
+                    var implicitIndexer = (BoundImplicitIndexerAccess)left;
+
+                    BoundExpression transformedImplicitIndexer;
+                    var argumentType = implicitIndexer.Argument.Type;
+                    if (TypeSymbol.Equals(argumentType, _compilation.GetWellKnownType(WellKnownType.System_Range), TypeCompareKind.ConsiderEverything))
```

Comment:
>TypeSymbol.Equals(argumentType, _compilation.GetWellKnownType(WellKnownType.System_Range), TypeCompareKind.ConsiderEverything) [](http://example.com/codeflow?start=24&length=126)

Consider the following simplification.
Check for ```System_Index``` instead and call ```TransformIndexPatternIndexerAccess``` for that case directly instead of doing that through ```TransformImplicitIndexerAccess```. Changes for ```TransformImplicitIndexerAccess``` could be reverted then.